src/analysis/special_functions/bernstein.lean

```diff
@@ -0,0 +1,301 @@
+  intros n nh npos',
+  have npos : 0 < (n:ℝ) := by exact_mod_cast npos',
+  -- Two easy inequalities we'll need later:
+  have w₁ : 0 ≤ 2 * ∥f∥ := mul_nonneg (by norm_num) (norm_nonneg f),
+  have w₂ : 0 ≤ 2 * ∥f∥ * δ^(-2 : ℤ) := mul_nonneg w₁ pow_minus_two_nonneg,
+  -- As `[0,1]` is compact, it suffices to check the inequality pointwise.
+  rw (continuous_map.norm_lt_iff _ h),
+  intro x,
+  -- The idea is to split up the sum over `k` into two sets,
+  -- `S`, where `x - k/n < δ`, and its complement.
+  let S := S f ε h n x,
+  calc
+    |(bernstein_approximation n f - f) x|
+        = |bernstein_approximation n f x - f x|
+                              : rfl
+    ... = |bernstein_approximation n f x - f x * 1|
+                              : by rw mul_one
+    ... = |bernstein_approximation n f x - f x * (∑ k : fin (n+1), bernstein n k x)|
+                              : by rw bernstein.probability
+    ... = |∑ k : fin (n+1), (f k/ₙ - f x) * bernstein n k x|
+                              : by simp [bernstein_approximation, finset.mul_sum, sub_mul]
+    ... ≤ ∑ k : fin (n+1), |(f k/ₙ - f x) * bernstein n k x|
+                              : finset.abs_sum_le_sum_abs
+    ... = ∑ k : fin (n+1), |f k/ₙ - f x| * bernstein n k x
+                              : by simp_rw [abs_mul, abs_eq_self.mpr bernstein_nonneg]
+    ... = ∑ k in S, |f k/ₙ - f x| * bernstein n k x +
+          ∑ k in Sᶜ, |f k/ₙ - f x| * bernstein n k x
+                              : (S.sum_add_sum_compl _).symm
+    -- We'll now deal with the terms in `S` and the terms in `Sᶜ` in separate calc blocks.
+    ... < ε/2 + ε/2 : add_lt_add_of_le_of_lt _ _
+    ... = ε : add_halves ε,
+    { -- We now work on the terms in `S`: uniform continuity and `bernstein.probability`
+      -- quickly give us a bound.
+      calc ∑ k in S, |f k/ₙ - f x| * bernstein n k x
+          ≤ ∑ k in S, ε/2 * bernstein n k x
+                                :  finset.sum_le_sum
+                                    (λ k m, (mul_le_mul_of_nonneg_right (le_of_lt (lt_of_mem_S m))
+                                      bernstein_nonneg))
+      ... = ε/2 * ∑ k in S, bernstein n k x
+                                : by rw finset.mul_sum
+      -- In this step we increase the sum over `S` back to a sum over all of `fin (n+1)`,
+      -- so that we can use `bernstein.probability`.
+      ... ≤ ε/2 * ∑ k : fin (n+1), bernstein n k x
+                                : mul_le_mul_of_nonneg_left
+                                    (finset.sum_le_univ_sum_of_nonneg (λ k, bernstein_nonneg))
+                                    (le_of_lt (half_pos h))
+      ... = ε/2 : by rw [bernstein.probability, mul_one] },
+      -- We now turn to working on `Sᶜ`: we control the difference term just using `∥f∥`,
+      -- and then insert a `δ^(-2) * (x - k/n)^2` factor
+      -- (which is at least one because we are not in `S`).
+      calc ∑ k in Sᶜ, |f k/ₙ - f x| * bernstein n k x
```

Comment:
Just to clarify, could you also enclose this calc block with braces (although I agree it is not strictly necessary)?

Reply:
Done.